streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -3616,33 +3616,33 @@ object ZStream extends ZStreamPlatformSpecificConstructors {
   def fromIterableM[R, E, O](iterable: ZIO[R, E, Iterable[O]]): ZStream[R, E, O] =
     fromEffect(iterable).mapConcat(identity)
 
-  def fromIterator[A](iterator: => Iterator[A]): ZStream[Any, Throwable, A] = {
-    object StreamEnd extends Throwable
-
-    ZStream.fromEffect(Task(iterator) <*> ZIO.runtime[Any]).flatMap { case (it, rt) =>
-      ZStream.repeatEffectOption {
-        Task {
-          val hasNext: Boolean =
-            try it.hasNext
-            catch {
-              case e: Throwable if !rt.platform.fatal(e) =>
-                throw e
-            }
-
-          if (hasNext) {
-            try it.next()
-            catch {
-              case e: Throwable if !rt.platform.fatal(e) =>
-                throw e
-            }
-          } else throw StreamEnd
-        }.mapError {
-          case StreamEnd => None
-          case e         => Some(e)
-        }
-      }
+  /**
+   * Creates a stream from an iterator that may throw exceptions.
+   */
+  def fromIterator[A](iterator: => Iterator[A], maxChunkSize: Int = 1): ZStream[Any, Throwable, A] =
```

Comment:
Just a heads up that adding a default parameter breaks backwards compatibility, like [here](https://github.com/zio/zio/pull/4205).

Reply:
Yes. I think since we are not guaranteeing binary compatibility at this point that is okay. I think this is the right API and it should be a very easy change, but open to other perspectives.

Reply:
You could also add another method with 2 parameters, and the 1 parameter one can delegate to this one.